simulator/func_sim/func_sim.cpp

```diff
@@ -75,6 +75,59 @@ void FuncSim<ISA>::run( const std::string& tr, uint64 instrs_to_run)
     }
 }
 
+template <typename ISA>
+void FuncSim<ISA>::gdb_load( const std::string &tr) {
+    mem->load_elf_file( tr);
+    sout << "MIPT-MIPS: Binary file " << tr << " loaded" << std::endl;
+    //TODO: print more useful info
+}
+
+template <typename ISA>
+void FuncSim<ISA>::gdb_prepare() {
+    PC = mem->startPC();
+    nops_in_a_row = 0;
+    sout << "MIPT-MIPS: prepared to run" << std::endl;
+    //TODO: print more useful info
+}
+
+template <typename ISA>
+void FuncSim<ISA>::gdb_resume( int steps) {
+    sout << "MIPT-MIPS: resuming, steps: " << steps << std::endl;
+    uint32 instrs_to_run = ((steps == 0) ? MAX_VAL32 : steps);
```

Comment:
If steps == 0. execution continues forever. Is it the behavior intended by GDB API?

Reply:
However, it is not an infinite loop, and MAX_VAL32 is not very capable for instruction numbers, only 4 seconds with IPC=1 and frequency = 1 GHz.